File: src/main/java/net/ndrei/teslacorelib/inventory/ColoredItemHandlerInfo.java
package net.ndrei.teslacorelib.inventory;

import net.minecraft.item.EnumDyeColor;

/**
 * Created by CF on 2016-12-17.
 */
public final class ColoredItemHandlerInfo {
    private String name;
    private EnumDyeColor color;
    private BoundingRectangle highlight;

    public ColoredItemHandlerInfo(String name, EnumDyeColor color) {
        this.name = name;
        this.color = color;
    }

    public ColoredItemHandlerInfo(String name, EnumDyeColor color, BoundingRectangle highlight) {
        this(name, color);
        this.highlight = highlight;
    }

    public EnumDyeColor getColor() {
        return this.color;
    }

    public String getName() {
        return this.name;
    }

    public BoundingRectangle getHighlight() { return this.highlight; }
}

File: src/main/kotlin/net/ndrei/teslacorelib/inventory/ColoredItemHandlerInfo.kt
package net.ndrei.teslacorelib.inventory

import net.minecraft.item.EnumDyeColor

@Suppress("unused")
/**
 * Created by CF on 2017-06-28.
 */
class ColoredItemHandlerInfo(val name: String, val color: EnumDyeColor, val highlight: BoundingRectangle, var index: Int = 0)